### PR TITLE
Implemented and added unit test for VR507-1

### PR DESCRIPTION
**Pull request by Mech0z**
## Description

For Tariffs 
- Resolution = PT1H Charge must contain 24 prices 
- Resolution = P1D Charge may only contain 1 price
- Resolution = PT15M Charge must contain 96 prices

## References

* #194

### CODE REVIEW

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+    [UnitTest]
+    public class ChargeTypeTariffPriceCountRuleTests
+    {
+        [Theory]
+        [InlineAutoMoqData(1, false)]
+        [InlineAutoMoqData(23, false)]
+        [InlineAutoMoqData(24, true)]
+        [InlineAutoMoqData(25, false)]
+        [InlineAutoMoqData(96, false)]
+        public void ChargePriceCountRule_WhenCalledWithPT1H_Expects24PricePoints(
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs:35]: I still think you should adhere to our naming convention _MemberName_WhenSomething_IsOrDoes_.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+        public void ChargePriceCountRule_WhenNotTariff_ShouldParseValidation(
+            ChargeType chargeType,
+            [NotNull] ChargeCommand chargeCommand)
+        {
+            chargeCommand.ChargeOperation.Type = chargeType;
+            var sut = new ChargeTypeTariffPriceCountRule(chargeCommand);
+            sut.IsValid.Should().BeTrue();
+        }
+
+        private static List<Point> GeneratePricePointList(Point point, int itemsInList)
```
Comment (BjarkeMeier) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs:116]:
```suggestion
        private static List<Point> GeneratePricePointList(Point item, int length)
```

**Review by BjarkeMeier — APPROVED** [review]
LGTM, except for some naming comments.

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRule.cs [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+using GreenEnergyHub.Charges.Domain.ChangeOfCharges.Transaction;
+
+namespace GreenEnergyHub.Charges.Application.Validation.InputValidation.ValidationRules
+{
+    public class ChargeTypeTariffPriceCountRule : IValidationRule
```
Comment (bemwen) [inline source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRule.cs:19]: @prtandrup This rule will not work with the new CIM model. We need to go through validations rules to find out which needs to be revised

**Review by bemwen — APPROVED** [review]

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs (hunk 2 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs]: Changed in 163c505020edda5a6f4eda5f1388b7c82dcf9925

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs (hunk 1 above)
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Tests/Application/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRuleTests.cs]: Changed names in 67fe3d310c674663ff51340c1b36704936391d2f

File: source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRule.cs [hunk 4]
```diff
@@ -0,0 +1,54 @@
...
+                {
+                    switch (_chargeCommand.ChargeOperation.Resolution)
+                    {
+                        case Resolution.PT15M:
+                            return _chargeCommand.ChargeOperation.Points.Count == PricePointsRequiredInPt15M;
+                        case Resolution.PT1H:
+                            return _chargeCommand.ChargeOperation.Points.Count == PricePointsRequiredInPt1H;
+                        case Resolution.P1D:
+                            return _chargeCommand.ChargeOperation.Points.Count == PricePointsRequiredInP1D;
+                    }
```
[the remark this answers is not in the document]
Comment (Mech0z) [reply source/GreenEnergyHub.Charges/source/GreenEnergyHub.Charges.Application/Validation/InputValidation/ValidationRules/ChargeTypeTariffPriceCountRule.cs]: Added in ea98698a14b2b36e42cfea615e914f6b6260aca9

**Pull request merged**